crates/sui-core/src/authority.rs

```diff
@@ -2129,6 +2143,13 @@ impl AuthorityState {
     }
 
     fn verify_narwhal_transaction(&self, certificate: &CertifiedTransaction) -> SuiResult {
+        // Ensure the input is a shared object certificate. Remember that Byzantine authorities
+        // may input anything into consensus.
+        fp_ensure!(
+            certificate.contains_shared_object(),
+            SuiError::NotASharedObjectTransaction
+        );
+
```

Comment:
Doesn't this check conflict with our checkpoint mechanism (which involves sequencing owned-object transactions through consensus)?